src/lib.rs

```diff
@@ -100,48 +102,21 @@ fn args_are_cpp(clang_args: &[String]) -> bool {
         .any(|w| w[0] == "-x=c++" || w[1] == "-x=c++" || w == &["-x", "c++"]);
 }
 
-/// A type used to indicate which kind of items do we have to generate.
-///
-/// TODO(emilio): Use `bitflags!`
-#[derive(Debug, Clone)]
-pub struct CodegenConfig {
-    /// Whether to generate functions.
-    pub functions: bool,
-    /// Whether to generate types.
-    pub types: bool,
-    /// Whether to generate constants.
-    pub vars: bool,
-    /// Whether to generate methods.
-    pub methods: bool,
-    /// Whether to generate constructors.
-    pub constructors: bool,
-    /// Whether to generate destructors.
-    pub destructors: bool,
-}
-
-impl CodegenConfig {
-    /// Generate all kinds of items.
-    pub fn all() -> Self {
```

Comment:
These methods should be kept, otherwise you can no longer say "generate everything except" via the API without enumerating everything.

Right now you can do:

```rust
CodegenConfig {
    vars: false,
    .. CodegenConfig::all(),
}
```

And there'd be no equivalent way of doing the same without listing everything.

Reply:
My understanding is that it is not possible to implement this because it uses the struct update syntax, the struct generated by bitfields only has one field (self.bits). It would possible to write your example as follows:

```rust
CodegenConfig::all().remove(CodegenConfig::VARS)
```